File: webui/src/ws_events.ts
export const WS_CONNECTED = "connected" as const;
export const WS_PING = "ping" as const;
export const WS_DASHBOARD_UPDATE = "dashboard_update" as const;
export const WS_LOGS_UPDATE = "logs_update" as const;
export const WS_LOGS_APPEND = "logs_append" as const;
export const WS_SCAN_PROGRESS = "scan_progress" as const;

export type WSEventType =
  | typeof WS_CONNECTED
  | typeof WS_PING
  | typeof WS_DASHBOARD_UPDATE
  | typeof WS_LOGS_UPDATE
  | typeof WS_LOGS_APPEND
  | typeof WS_SCAN_PROGRESS;

File: pt_invite_watcher/ws_events.py
from __future__ import annotations

"""
WebSocket event type constants.

Wire protocol must remain stable; these values are the canonical strings used by
the WebUI and any external clients.
"""

WS_CONNECTED = "connected"
WS_PING = "ping"
WS_DASHBOARD_UPDATE = "dashboard_update"
WS_LOGS_UPDATE = "logs_update"
WS_LOGS_APPEND = "logs_append"
WS_SCAN_PROGRESS = "scan_progress"

__all__ = [
    "WS_CONNECTED",
    "WS_PING",
    "WS_DASHBOARD_UPDATE",
    "WS_LOGS_UPDATE",
    "WS_LOGS_APPEND",
    "WS_SCAN_PROGRESS",
]

